application/client/src/app/ui/elements/recent/state.ts

```diff
@@ -47,13 +49,15 @@ export class State {
     }
 
     public filtering() {
-        this.actions.forEach((action) => {
-            action.filter(this.filter.value());
-        });
+        this.matcher.search(this.filter.value());
+        this.actions.sort((a: WrappedAction, b: WrappedAction) => b.getScore() - a.getScore());
         this.update.emit();
     }
 
     public getFilteredActions(): WrappedAction[] {
-        return this.actions.filter((a) => a.filtered);
+        if (this.filter.value() === '') {
+            return this.actions;
+        }
```

Comment:
if score would be inited (on rust level or even on abstract level), we don't need to check filter value